chain/near/src/trigger.rs

```diff
@@ -480,7 +480,17 @@ mod tests {
                 )));
             }
 
-            return Ok(Vec::from(&self.memory[start_offset..end_offset_exclusive]));
+            let src = &self.memory[start_offset..end_offset_exclusive];
+
+            // See also 6d2f040c-9361-4cf9-8b3c-426f87cc0b21
+            unsafe {
+                let uninit_src: &[MaybeUninit<u8>] = std::mem::transmute(src);
```

Comment:
Could this block be achieved without unsafe? 

From the docs: 
```
transmute is incredibly unsafe. There are a vast number of ways to cause [undefined behavior](https://doc.rust-lang.org/stable/reference/behavior-considered-undefined.html) with this function. transmute should be the absolute last resort.
```

Reply:
Not really. Working with `MaybeUninit` infers that there will be unsafety somewhere. And though `transmute` comes with strong warnings, it is currently the only way to work with `MaybeUninit` effectively. There are some nightly APIs that will simplify our use of unsafe, like [write_slice](https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html#method.write_slice) but for now we have to copy-paste the internals of those APIs.